Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -40,15 +40,21 @@ private static class NativeMethods
             [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
             public delegate void SessionWaitCallback(IntPtr task_completion_source, int error_code, PrimitiveValue message);
 
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            public delegate void SessionConnectionChangeCallback(IntPtr connectionChangeCallbackHandle);
```

Comment:
Let's call this `SessionPropertyChangedCallback` and have it pass the property name as well. This way adding state notifications will be fairly trivial.